client/src/components/hero/Hero.tsx

```diff
@@ -17,64 +20,178 @@ interface HomeProps {
  * Hero for main home page.
  */
 const UnstyledHero: FunctionComponent<HomeProps> = ({ className }) => {
+  const { viewer } = useContext(ViewerContext);
+  // const announcementRef = useRef(0);
+
+  const [mainAnnouncementPosition, setMainAnnouncementPosition] = useState(0);
```

Comment:
I think `mainAnnouncementIdx` makes more sense. this is not really a position it's index of what is shown as main post.